src/Algebra/Graph/HigherKinded/Class.hs

```diff
@@ -212,7 +212,9 @@ edge x y = connect (vertex x) (vertex y)
 -- 'vertexSet'   . vertices == Set.'Set.fromList'
 -- @
 vertices :: Graph g => [a] -> g a
-vertices = overlays . map vertex
```

Comment:
I wonder if we can achieve the same speedup using rewrite rules? What if we add a function `overlaysMap` which will be implemented as you define below (but with an arbitrary function `f` instead of `vertex`) and then add a rewrite rule to optimise `overlays . map` to `overlaysMap`?